### PR TITLE
Add option to request manual quit on tui

### PR DESCRIPTION
**Pull request by vincentmasse**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Related Issues/PRs

#2233 

### Changes

Add option to learner builder to wait for user to quit the renderer.

### Testing

Manual test, TUI waits for q to close.

### CODE REVIEW

**Comment by vincentmasse** [comment]
I wasn't sure about the best way to implement this, particularly if adding it to the builder was the right approach. Let me know if you have any suggestions.

**Comment by vincentmasse** [comment]
image
I got this error while running run_checks. I opened the PR since it's not linked with the changes I made. I suspect it's because the intel graphics used isn't implementing cooperative matrix multiplication.

File: crates/burn-train/src/renderer/tui/renderer.rs [hunk 1]
```diff
@@ -116,9 +117,15 @@ impl TuiMetricsRenderer {
             interuptor,
             popup: PopupState::Empty,
             previous_panic_hook: Some(previous_panic_hook),
+            manual_quit: false,
         }
     }
 
+    /// Enable manual quit after training.
+    pub fn enable_manual_quit(&mut self) {
```
Comment (laggui) [inline crates/burn-train/src/renderer/tui/renderer.rs:125]: I think I would rename the method and take ownership to return the renderer. That way, we can keep the flow similar to the builder when declaring the renderer, e.g. `TuiMetricsRenderer::new(...).persistent()`

```rust
pub fn persistent(mut self) -> Self {
    self.manual_quit = true;
    self
}
```

the `manual_quit` field could also be renamed but not too critical.

**Review by laggui — COMMENTED** [review]
Ah that's pretty good!

I think we could make one minimal change, after which the usage would look something like this:

```rust
let learner = LearnerBuilder::new(ARTIFACT_DIR)
    // ...
    .devices(vec![device.clone()])
    .num_epochs(config.num_epochs)
    .summary();

let tui = TuiMetricsRenderer::new(learner.interrupter(), None).persistent();
let learner = learner
    .renderer(tui)
    .build(Model::new(&device), config.optimizer.init(), 1e-4);
```

While it "breaks" the builder pattern, I think it's perfectly fine for now. We could eventually improve this but I don't think it's a critical issue atm.

**Review by laggui — APPROVED** [review]
Thanks 🎉

**Pull request merged**